mempool/src/pool.rs

```diff
@@ -0,0 +1,2768 @@
+use std::cell::Cell;
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use mockall::*;
+use parity_scale_codec::Encode;
+
+use common::chain::transaction::Transaction;
+use common::chain::transaction::TxInput;
+use common::chain::OutPoint;
+use common::primitives::amount::Amount;
+use common::primitives::Id;
+use common::primitives::Idable;
+use common::primitives::H256;
+
+use logging::log;
+
+use utils::newtype;
+
+use crate::error::Error;
+use crate::error::TxValidationError;
+use crate::feerate::FeeRate;
+use crate::feerate::INCREMENTAL_RELAY_FEE_RATE;
+
+const ROLLING_FEE_BASE_HALFLIFE: Time = 60 * 60 * 12;
+// TODO this willbe defined elsewhere (some of limits.rs file)
+const MAX_BLOCK_SIZE_BYTES: usize = 1_000_000;
+
+const MAX_BIP125_REPLACEMENT_CANDIDATES: usize = 100;
+
+// TODO this should really be taken from some global node settings
+const RELAY_FEE_PER_BYTE: usize = 1;
+
+const MAX_MEMPOOL_SIZE_BYTES: usize = 3_000_000;
```

Comment:
Our mempool is quite small in comparison to our block

Reply:
Took this number from Bitcoin Core too. I don't know, does this mean our mempool is too small or that our max block is too big?

Reply:
You sure?

`/** Default for -maxmempool, maximum megabytes of mempool memory usage */

static const unsigned int DEFAULT_MAX_MEMPOOL_SIZE = 300;`

I make ours 100 times smaller, assuming 1000^2 bytes rather than 2^20 bytes.

Reply:
A 3MB mempool is not enough for a 1MB blocksize. You could argue since our blocks are produced 10 times as frequently we could reduce it from 300MB but not by a factor of 100 in my opinion.

Reply:
I got confused by a factor of a 100 :sweat_smile: